src/System.Runtime/tests/System/Threading/WaitHandle.cs

```diff
@@ -43,6 +43,16 @@ public static void WaitAny()
     }
 
     [Fact]
+    public static void WaitAnySameHandles()
+    {
+        ManualResetEvent[] wh = new ManualResetEvent[2];
+        wh[0] = new ManualResetEvent(true);
+        wh[1] = wh[0];
+
+        Assert.Equal(WaitHandle.WaitAny(wh), 0);
```

Comment:
expected argument should be passed first

    Assert.Equal(0, WaitHandle.WaitAny(wh));